core/commands/partials/terraform_configurator.rb

```diff
@@ -74,73 +78,86 @@ def configure(node)
+  # @return [Result::Base]
+  def up_machines(nodes)
+    TerraformService.resource_type(@config.provider).and_then do |resource_type|
+      TerraformService.init(@ui, @config.path)
+      target_nodes = TerraformService.nodes_to_resources(nodes, resource_type)
+      @attempts.times do |attempt|
+        @ui.info("Up nodes #{nodes}. Attempt #{attempt + 1}.")
+        destroy_nodes(target_nodes.keys) if @recreate_nodes || attempt.positive?
+
+        apply_result = TerraformService.apply(target_nodes.values, @ui, @config.path)
+        return Result.ok('') if apply_result.success?
 
-  def node_running?(node)
-    result = TerraformService.resource_type(@config.provider).and_then do |resource_type|
-      return TerraformService.resource_running?(resource_type, node, @ui, @config.path)
+        TerraformService.running_resources(@ui, @config.path).and_then do |running_resources|
+          target_nodes = target_nodes.reject { |_node, resource| running_resources.include?(resource) }
+        end
+      end
+      Result.error("Error up of machines: #{target_nodes.keys}")
     end
-    @ui.error(result.error)
-    false
   end
 
-  # Create and configure node, or recreate if it needs to fix.
+  # Configure machine via mdbci.
   #
-  # @param node [String] name of node which needs to be configured
-  # @return [Bool] configuration result
-  def bring_up_and_configure(node)
+  # @param node [String] name of node to configure
+  # @return [Hash] result of configuring in format { node: String, result: Boolean, logger: Out }
+  def configure_machine(node)
+    logger = retrieve_logger_for_node
+    configure_result = false
     @attempts.times do |attempt|
-      @ui.info("Bring up and configure node #{node}. Attempt #{attempt + 1}.")
-      bring_up_result = if @recreate_nodes || attempt.positive?
-                          destroy_node(node)
-                          bring_up_machine(node)
-                        elsif !node_running?(node)
-                          bring_up_machine(node)
-                        end
-      break if !bring_up_result.nil? && bring_up_result.error?
-      next unless node_running?(node)
+      @ui.info("Configure node #{node}. Attempt #{attempt + 1}.")
+      configure_result = configure(node, logger)
+      break if configure_result
+    end
+    { node: node, result: configure_result, logger: logger }
+  end
 
-      store_network_settings(node).and_then do
-        return true if configure(node)
-      end
+  # Configure machines via mdbci.
+  #
+  # @param nodes [Array<String>] name of nodes to configure
+  # @return [Result::Base]
+  def configure_machines(nodes)
+    @ui.info("Configure machines: #{nodes}")
+    error_network_nodes = nodes.map { |node| store_network_settings(node, @ui) }.reject(&:success?)
+    unless error_network_nodes.empty?
+      @ui.error("Error of storing network settings for nodes: #{error_network_nodes}")
+      return Result.error(error_network_nodes)
```

Comment:
If we can not determine the IPs of the machines, should not we try to recreate them and get them once again? I think that this step should be moved to the `up_machines` as the last step of configuration.